clang/lib/3C/unittests/ConstraintTests.cpp

```diff
@@ -0,0 +1,267 @@
+#include "gtest/gtest.h"
```

Comment:
Have you resurrected this test file? I think Matt might have dropped it. Same with the changes to CMakeLists.txt

Reply:
Indeed.  As we know, this kind of thing happened before in #330 (although probably with a different cause) and can be very disruptive, so I'd like to track down and eliminate the causes whenever possible.  Aaron, it looks like you did something unusual in d9b5da666f3403423565f512ef536fad6640d781: what and why?  Then when I try to reproduce 7e456ce9ca9c664b30bfc8b2c5421d28d2732d2f by running `git merge` myself, there are rename/delete conflicts on the files that ended up incorrectly resurrected.  Unfortunately, the way git handles that situation, by leaving a stage-2 entry in the index, is prone to exactly the problem we saw here if the user runs `git add` without paying attention.  I consider this a defect in git, but I'm not aware of an easy workaround and I doubt it's worth trying to develop one right now.